subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -772,25 +781,20 @@ const struct bt_mesh_model_cb _bt_mesh_scene_srv_cb = {
 static int scene_setup_srv_init(struct bt_mesh_model *model)
 {
 	struct bt_mesh_scene_srv *srv = model->user_data;
+	int err;
 
 	if (!srv) {
 		return -EINVAL;
 	}
 
 	srv->setup_mod = model;
 
-	/* Model extensions:
-	 * To simplify the model extension tree, we're flipping the
-	 * relationship between the scene server and the scene
-	 * setup server. In the specification, the scene setup
-	 * server extends the scene server, which is the opposite of
-	 * what we're doing here. This makes no difference for the mesh
-	 * stack, but it makes it a lot easier to extend this model, as
-	 * we won't have to support multiple extenders.
-	 */
-	bt_mesh_model_extend(srv->model, srv->setup_mod);
+	err = bt_mesh_model_extend(srv->setup_mod, srv->dtt_srv->model);
```

Comment:
Needs a check that `srv->dtt_srv->model` is not NULL as DTT may be (but must not) instantiated after Scene server model.

Reply:
`srv->dtt_srv->model` -> `dtt_srv->model`.